src/git.rs

```diff
@@ -0,0 +1,300 @@
+use rover_client::query::{graph, subgraph};
+#[derive(Debug)]
+pub struct GitContext {
+    pub branch: Option<String>,
+    pub committer: Option<String>,
+    pub commit: Option<String>,
+    pub message: Option<String>,
+    pub remote_url: Option<String>,
+}
+
+type GraphPushContextInput = graph::push::push_schema_mutation::GitContextInput;
+impl Into<GraphPushContextInput> for GitContext {
+    fn into(self) -> GraphPushContextInput {
+        GraphPushContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type GraphCheckContextInput = graph::check::check_schema_query::GitContextInput;
+impl Into<GraphCheckContextInput> for GitContext {
+    fn into(self) -> GraphCheckContextInput {
+        GraphCheckContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type SubgraphPushContextInput = subgraph::push::push_partial_schema_mutation::GitContextInput;
+impl Into<SubgraphPushContextInput> for GitContext {
+    fn into(self) -> SubgraphPushContextInput {
+        SubgraphPushContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type SubgraphCheckContextInput = subgraph::check::check_partial_schema_query::GitContextInput;
+impl Into<SubgraphCheckContextInput> for GitContext {
+    fn into(self) -> SubgraphCheckContextInput {
+        SubgraphCheckContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
```

Comment:
really nice to bring this out into this file so it's hidden - i'd love if we could put it at the bottom of the file though since it's just boilerplate

Reply:
I'll move it to right above the tests. I feel like the very bottom would feel weird to me